Northstar.Custom/mod/scripts/vscripts/gamemodes/_gamemode_fw.nut

```diff
@@ -0,0 +1,2081 @@
+        {
+            if( lastDamagedTime + FW_TURRET_DAMAGED_DEBOUNCE >= Time() ) // recent underattack
+            {
+                if( turret.GetShieldHealth() > 0 ) // has shields
+                    stateFlag = TURRET_SHIELDED_UNDERATTACK_MLT_FLAG
+                else
+                    stateFlag = TURRET_UNDERATTACK_MLT_FLAG
+
+                // these dialogue have 30s debounce inside
+                if( isBaseTurret )
+                    PlayFactionDialogueToTeam( "fortwar_baseTurretsUnderAttack", TEAM_MILITIA )
+                else
+                    PlayFactionDialogueToTeam( "fortwar_awayTurretsUnderAttack", TEAM_MILITIA )
+            }
+            else if( turret.GetShieldHealth() > 0 ) // has shields left
+                stateFlag = TURRET_SHIELDED_MLT_FLAG
+            else
+                stateFlag = TURRET_MLT_FLAG
+        }
+
+        // natural states
+        if( iconTeam == TEAM_UNASSIGNED )
+        {
+            if( lastDamagedTime + FW_TURRET_DAMAGED_DEBOUNCE >= Time() ) // recent underattack
+                stateFlag = TURRET_UNDERATTACK_NATURAL_FLAG
+            else
+                stateFlag = TURRET_NATURAL_FLAG
+        }
+
+        SetGlobalNetInt( stateVarName, stateFlag )
+
+        // update these
+        lastFrameTeam = turretTeam
+        lastFrameIsAlive = turretAlive
+
+        WaitFrame()
+    }
+
+}
+
+////////////////////////////////
+///// TURRET FUNCTIONS END /////
+////////////////////////////////
+
+
+
+///////////////////////////////
+///// HARVESTER FUNCTIONS /////
+///////////////////////////////
+
+void function startFWHarvester()
+{
+    thread HarvesterThink(fw_harvesterImc)
+	thread HarvesterAlarm(fw_harvesterImc)
+    thread UpdateHarvesterHealth( TEAM_IMC )
+
+    thread HarvesterThink(fw_harvesterMlt)
+	thread HarvesterAlarm(fw_harvesterMlt)
+    thread UpdateHarvesterHealth( TEAM_MILITIA )
+}
+
+entity function FW_GetTeamHarvesterProp( int team )
+{
+    if( team == TEAM_IMC )
+        return fw_harvesterImc.harvester
+    else if( team == TEAM_MILITIA )
+        return fw_harvesterMlt.harvester
+
+    unreachable // crash the game
+}
+
+void function FW_createHarvester()
+{
+    // mlt havester spawn
+    fw_harvesterImc = SpawnHarvester( file.harvesterImc_info.GetOrigin(), file.harvesterImc_info.GetAngles(), GetCurrentPlaylistVarInt( "fw_harvester_health", FW_DEFAULT_HARVESTER_HEALTH ), GetCurrentPlaylistVarInt( "fw_harvester_shield", FW_DEFAULT_HARVESTER_SHIELD ), TEAM_IMC )
+	fw_harvesterImc.harvester.Minimap_SetAlignUpright( true )
+	fw_harvesterImc.harvester.Minimap_AlwaysShow( TEAM_IMC, null )
+	fw_harvesterImc.harvester.Minimap_AlwaysShow( TEAM_MILITIA, null )
+	fw_harvesterImc.harvester.Minimap_SetHeightTracking( true )
+	fw_harvesterImc.harvester.Minimap_SetZOrder( MINIMAP_Z_OBJECT )
+	fw_harvesterImc.harvester.Minimap_SetCustomState( eMinimapObject_prop_script.FD_HARVESTER )
+    AddEntityCallback_OnDamaged( fw_harvesterImc.harvester, OnHarvesterDamaged )
```

Comment:
change this to `AddEntityCallback_OnPostDamaged` so other callbacks (like northstar railgun charge) can be executed first